nomt/src/beatree/ops/update/branch_updater.rs

```diff
@@ -407,79 +357,40 @@ impl BranchUpdater {
-        let mut split_point = 0;
-
-        let mut left_gauge = BranchGauge::new();
-        while left_gauge.body_size() < midpoint {
-            match *&self.ops[split_point] {
-                BranchOp::Insert(key, _) => {
-                    if left_gauge.body_size_after(key, separator_len(&key)) > BRANCH_NODE_BODY_SIZE
-                    {
-                        if left_gauge.body_size() < BRANCH_MERGE_THRESHOLD {
-                            // rare case: body was artifically small due to long shared prefix.
-                            // start applying items without prefix compression. we assume items are less
-                            // than half the body size, so the next item should apply cleanly.
-                            left_gauge.stop_prefix_compression();
-                        } else {
-                            break;
-                        }
-                    }
-                }
-                BranchOp::Update(pos, _) => {
-                    let key = self.base.as_ref().unwrap().key(pos);
-                    if left_gauge.body_size_after(key, separator_len(&key)) > BRANCH_NODE_BODY_SIZE
-                    {
-                        if left_gauge.body_size() < BRANCH_MERGE_THRESHOLD {
-                            // same case as above
-                            left_gauge.stop_prefix_compression();
-                        } else {
-                            break;
-                        }
-                    }
-                }
-                BranchOp::KeepChunk(..) => {
-                    // try to split the chunk to make it fit into the available space
-                    let n_items = split_keep_chunk(
-                        self.base.as_ref().unwrap(),
-                        &left_gauge,
-                        &mut self.ops,
-                        split_point,
-                        midpoint,
-                        BRANCH_NODE_BODY_SIZE,
-                    );
 
-                    if n_items == 0 {
-                        // if no item from the chunk is capable to fit then
-                        // try to extract the first element from the chunk and storing it
-                        // back as an Insert to check if `stop_prefix_compression` is required
-                        extract_insert_from_keep_chunk(
-                            self.base.as_ref().unwrap(),
-                            &mut self.ops,
-                            split_point,
-                        );
+        let (split_point, left_gauge) = match self.consume_and_update_until(0, midpoint) {
+            Ok((split_point, left_gauge)) => (split_point, left_gauge),
+            // If the current ops cannot reach the target and there is a cutoff
+            // return NeedsMerge with the relative cutoff.
+            Err(new_gauge) if self.cutoff.is_some() => {
+                self.gauge = new_gauge;
+                return DigestResult::NeedsMerge(self.cutoff.unwrap());
```

Comment:
UNWRAP!

Reply:
Wow, yes, I totally forgot to comment all unwraps :sweat_smile: